presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkInjectorFactory.java

```diff
@@ -50,25 +53,7 @@
     private final Optional<Map<String, String>> eventListenerProperties;
     private final SqlParserOptions sqlParserOptions;
     private final List<Module> additionalModules;
-    private final Optional<Module> accessControlModuleOverride;
-
-    public PrestoSparkInjectorFactory(
-            SparkProcessType sparkProcessType,
-            Map<String, String> configProperties,
-            Map<String, Map<String, String>> catalogProperties,
-            Optional<Map<String, String>> eventListenerProperties,
-            SqlParserOptions sqlParserOptions,
-            List<Module> additionalModules)
-    {
-        this(
-                sparkProcessType,
-                configProperties,
-                catalogProperties,
-                eventListenerProperties,
-                sqlParserOptions,
-                additionalModules,
-                Optional.empty());
-    }
+    private final boolean isForTesting;
```

Comment:
I'm not a big fan of this change. `isForTesting` seems to be too generic, and it is hard to understand what does it mean exactly. Also the `TestingAccessControlModule ` is something very `AbstractTestQueries` specific, it seems a little weird to have it initialized here. Although `accessControlModuleOverride` doesn't seem to be the cleanest solution either, it seems to be a lesser evil. I would recommend to keep the `accessControlModuleOverride` as is.

Reply:
@arhimondr : I have different opinion. `accessControlModuleOverride` sounds too generic to me, and the actual use case is very limited for `TestingControlModule` . And it looks to me a very generic overridden mechanism is introduced for a very specific use case .

In Presto main code base, I see we have `PrestoServer` and `TestingPrestoServer`.

Reply:
The `QueryRunner` interface is expected to return `TestingAccessControlManager`: https://github.com/prestodb/presto/blob/master/presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java#L351

It is done by getting the specific `TestingAccessControlManager` instance from the context: https://github.com/prestodb/presto/blob/master/presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java#L233

That's why the `TestingAccessControlManager` is passed to the InjectorFactory explicitly as an override: https://github.com/prestodb/presto/blob/master/presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java#L210

It feels a little not intuitive that if the `isForTesting` is set to `true` that specific implementation of `AccessControlManager` is going to be bound. As always it is better to me more explicit than less explicit. 

Similarly in the `TestingPrestoServer` it is bound explicitly: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/server/testing/TestingPrestoServer.java#L253, same as in the LocalQueryRunner: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/testing/LocalQueryRunner.java#L365

Approaching from the other side - the `PrestoSparkInjectorFactory` is a generic factory. It shouldn't encapsulate `AbstractTestQueries` specific details. Setting up some components internally in the `PrestoSparkInjectorFactory` specifically for the `AbstractTestQueries` breaks the encapsulation. The `PrestoSparkInjectorFactory` should know nothing about the `AbstractTestQueries`.

Also `PrestoSparkInjectorFactory` is used for integration testing as well, but in that case the real `AccessControlManager` is used, which might sound counter intuitive why the `isForTesting` flag is set to `false` in that case.